### PR TITLE
Incrementally add city polygons for CountMap

### PR DESCRIPTION
**Pull request by tejiaz**
- Fix the "blinking issue" caused by "removeLayer()" and "addLayer()" at the city level for CountMap
- Incrementally add city polygons, rather than calling these two functions.

### CODE REVIEW

File: examples/twittermap/web/public/javascripts/map/controllers.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
     $scope.doNormalization = false;
     $scope.doSentiment = false;
     $scope.infoPromp = config.mapLegend;
+    $scope.addCityLayerRequest = false;
```
Comment (waans11) [inline examples/twittermap/web/public/javascripts/map/controllers.js]: Let's put a comment about this variable before declaring this. What does it do and why it is needed.

File: examples/twittermap/web/public/javascripts/map/controllers.js [hunk 2]
```diff
@@ -316,6 +321,7 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
             }
             if($scope.polygons.cityPolygons) {
               $scope.map.removeLayer($scope.polygons.cityPolygons);
+              $scope.polygons.cityPolygons = null;
```
Comment (waans11) [inline examples/twittermap/web/public/javascripts/map/controllers.js]: Is this necessary?

File: examples/twittermap/web/public/javascripts/map/controllers.js [hunk 3]
```diff
@@ -332,6 +338,7 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
...
```
Comment (waans11) [inline examples/twittermap/web/public/javascripts/map/controllers.js]: Is this necessary?

File: examples/twittermap/web/public/javascripts/map/controllers.js (hunk 1 above)
Comment (waans11) [reply examples/twittermap/web/public/javascripts/map/controllers.js]: And why don't we set it to true as default?

File: examples/twittermap/web/public/javascripts/map/controllers.js [hunk 4]
```diff
@@ -683,23 +690,56 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
...
                 if (!$scope.status.init) {
-                    resetGeoIds($scope.bounds, $scope.geojsonData.city, 'cityID');
+                    resetGeoIds($scope.bounds, data, 'cityID');
                     cloudberry.parameters.geoLevel = 'city';
                     cloudberry.query(cloudberry.parameters);
                 }
                 resetGeoInfo("city");
-                $scope.map.addLayer($scope.polygons.cityPolygons);
+
+                // incrementally add polygons after user zoom in
```
Comment (waans11) [inline examples/twittermap/web/public/javascripts/map/controllers.js]: after user moves to a new region?

File: examples/twittermap/web/public/javascripts/map/controllers.js [hunk 5]
```diff
@@ -683,23 +690,56 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common','clou
...
+                if($scope.addCityLayerRequest === false){
+                    var diffData = [];
+
+                    // finding out the new incoming cities which are not stored in geojsonData.city yet
+                    for(i =0; i < data.features.length; i++){
+                        var overlap = false;
+
+                        for(j =0; j < $scope.geojsonData.city.features.length; j++){
```
Comment (waans11) [inline examples/twittermap/web/public/javascripts/map/controllers.js]: Can we make a set that includes "$scope.geojsonData.city.features[j].properties.cityID" and use it like putValues() in mapResultCache?

**Comment by chenlica** [comment]
Add a detailed description of the previous "bug," and the way we fix it.  The current description is too brief.

**Pull request merged**